File: api/src/main/java/graphql/nadel/util/OperationNameUtil.java
package graphql.nadel.util;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class OperationNameUtil {
    private OperationNameUtil() {
    }

    /**
     * This is deprecated because we are moving away from this nadel_2_service naming scheme. We are
     * just forwarding operation names in the future. But for now, we need to support it for migration
     * purposes.
     */
    @Deprecated
    public static String getLegacyOperationName(@NotNull String serviceName, @Nullable String originalOperationName) {
        final var baseName = "nadel_2_" + serviceName;
        if (originalOperationName != null) {
            return baseName + "_" + originalOperationName;
        } else {
            return baseName;
        }
    }
}

File: api/src/main/java/graphql/nadel/util/OperationNameUtil.kt
package graphql.nadel.util

// todo: make internal once we merge api/ and engine-nextgen
object OperationNameUtil {
    /**
     * This is deprecated because we are moving away from this nadel_2_service naming scheme. We are
     * just forwarding operation names in the future. But for now, we need to support it for migration
     * purposes.
     */
    @Deprecated("")
    fun getLegacyOperationName(serviceName: String, originalOperationName: String?): String? {
        val baseName = "nadel_2_$serviceName"
        return if (originalOperationName != null) {
            baseName + "_" + originalOperationName
        } else {
            baseName
        }
    }
}
